crates/hyperdrive-math/src/lp.rs

```diff
@@ -600,6 +621,567 @@ impl State {
+        // minimum share reserves with a large value of k.
+        if ending_present_value >= starting_present_value {
+            return Ok(fixed!(0));
+        }
+
+        // NOTE: This subtraction is safe since the ending present value is less
+        // than the starting present value and the rhs is rounded down.
+        //
+        // Calculate the amount of withdrawal shares that can be redeemed.
+        let lp_total_supply = active_lp_total_supply + withdrawal_shares_total_supply;
+        Ok(lp_total_supply
+            - lp_total_supply.mul_div_down(ending_present_value, starting_present_value))
+    }
+
+    /// One of the edge cases that occurs when using Newton's method for
+    /// the share proceeds while distributing excess idle is when the net
+    /// curve trade is larger than the max bond amount. In this case, the
+    /// the present value simplifies to the following:
+    ///
+    /// PV(dz) = (z - dz) + net_c(dz) + net_f - z_min
+    ///        = (z - dz) - z_max_out(dz) + net_f - z_min
+    ///
+    /// There are two cases to evaluate:
+    ///
+    /// (1) zeta > 0:
+    ///
+    /// z_max_out(dz) = ((z - dz) / z) * (z - zeta) - z_min
+    ///
+    /// =>
+    ///
+    /// PV(dz) = zeta * ((z - dz) / z) + net_f
+    ///
+    /// (2) zeta <= 0:
+    ///
+    /// z_max_out(dz) = (z - dz) - z_min
+    ///
+    /// =>
+    ///
+    /// PV(dz) = net_f
+    ///
+    /// Since the present value is constant with respect to the share
+    /// proceeds in case 2, Newton's method has achieved a stationary point
+    /// and can't proceed. On the other hand, the present value is linear
+    /// with respect to the share proceeds, and we can solve for the next
+    /// step of Newton's method directly as follows:
+    ///
+    /// PV(0) / l = PV(dz) / (l - w)
+    ///
+    /// =>
+    ///
+    /// dz = z - ((PV(0) / l) * (l - w) - net_f) / (zeta / z)
+    ///
+    /// We round the share proceeds down to err on the side of the
+    /// withdrawal pool receiving slightly less shares.
+    fn calculate_distribute_excess_idle_share_proceeds_net_long_edge_case(
+        &self,
+        original_share_adjustment: I256,
+        original_share_reserves: FixedPoint,
+        starting_present_value: FixedPoint,
+        active_lp_total_supply: FixedPoint,
+        withdrawal_shares_total_supply: FixedPoint,
+    ) -> Result<FixedPoint> {
+        // If the original share adjustment is zero or negative, we cannot
+        // calculate the share proceeds. This should never happen, but for
+        // safety we return a failure flag and break the loop at this point.
+        if original_share_adjustment <= I256::zero() {
+            return Err(eyre!("Original share adjustment is zero or negative."));
+        }
+        // Calculate the net flat trade.
+        let net_flat_trade = self.calculate_net_flat_trade(
+            self.long_average_maturity_time(),
+            self.short_average_maturity_time(),
+        )?;
```

Comment:
Shouldn't this be `time_remaining` instead of `maturity_time`?

Reply:
yes! good catch.  I guess this passed because the timestamp was always zero in the tests.